### PR TITLE
Merge `main-tokens` into `main`

### PR DESCRIPTION
**Pull request by mischreiber**
### Platforms Impacted
- [x] iOS
- [x] macOS

### Description of changes

Merges `main-tokens` into `main`.

All changes/diffs are identical to those in #950:
- #950 

One additional change: `MicrosoftFluentUI.podspec` has been updated with two changes:
- Restore version number to `0.4.9`.
- Update Swift version to `5.6`, used in Xcode 13.3.

### Verification

See notes in #950. Additional testing has been limited to basic sanity pass (build, unit tests, `pod lib lint`, basic sanity UI pass).

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/974)

### CODE REVIEW

File: MicrosoftFluentUI.podspec [hunk 1]
```diff
@@ -1,12 +1,12 @@
 Pod::Spec.new do |s|
   s.name             = 'MicrosoftFluentUI'
-  s.version          = '0.4.9'
+  s.version          = '0.5.0'
```
Comment (huwilkes) [inline MicrosoftFluentUI.podspec:3]: This is pretty much the only change from the main-tokens branch, right?

Comment (huwilkes) [reply MicrosoftFluentUI.podspec:3]: Also, should this be changed to 5.0 now, or should there be a change just after this to bump this and the rest of the version bump info?

File: ios/FluentUI/Obscurable/DimmingView.swift [hunk 2]
```diff
@@ -18,13 +18,34 @@ public enum DimmingViewType: Int {
...
     }
 
     private var type: DimmingViewType
 
+    /// override default black dimmed color
+    public var dimmedBlackColor: UIColor? {
+        didSet {
+            setBackground(type: type)
+        }
+    }
```
Comment (huwilkes) [inline ios/FluentUI/Obscurable/DimmingView.swift:33]: Likely mostly harmless, but it seems like this was part of the change to add tokens to the drawer, #476 , which hasn't made its way in to main yet. So, I don't think any of the optional colors added here would actually be used in the DrawerPresentationController.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorTokensDemoController.swift [hunk 3]
```diff
@@ -0,0 +1,214 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import FluentUI
+import UIKit
+
+class ColorTokensDemoController: DemoTableViewController {
```
Comment (rdeassis) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorTokensDemoController.swift:9]: It seems like this DemoController is not being included in the Demo app.

File: MicrosoftFluentUI.podspec (hunk 1 above)
[the remark this answers is not in the document]
Comment (mischreiber) [reply MicrosoftFluentUI.podspec:3]: Sure, I can do that.

Comment (mischreiber) [reply MicrosoftFluentUI.podspec:3]: Yep @rdeassis mentioned that too. I'll do that separately.

File: ios/FluentUI.Demo/FluentUI.Demo/Resources/Assets.xcassets/LeftNav/Contents.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
[the remark this answers is not in the document]
Comment (mischreiber) [reply ios/FluentUI.Demo/FluentUI.Demo/Resources/Assets.xcassets/LeftNav/Contents.json:1]: Good catch, I'll remove those.

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorTokensDemoController.swift (hunk 3 above)
Comment (mischreiber) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorTokensDemoController.swift:9]: True! I'll add it.

File: ios/FluentUI/Obscurable/DimmingView.swift (hunk 2 above)
Comment (mischreiber) [reply ios/FluentUI/Obscurable/DimmingView.swift:33]: Hmm I'll take a look.

File: MicrosoftFluentUI.podspec (hunk 1 above)
[the remark this answers is not in the document]
Comment (mischreiber) [reply MicrosoftFluentUI.podspec:3]: Yep I'll do that separately all at once.

**Review by huwilkes — APPROVED** [review]

**Review by rdeassis — APPROVED** [review]

**Pull request merged**